### ISSUE TITLE
one route with limit

### ISSUE THREAD

**Issue by borodadada** [fastify/fastify-rate-limit#76]
Hello, i try use limit on one route but not work
example

```
fastifyGlobal.register(require('fastify-rate-limit'),
{
    global : false, // default true
    whitelist: [], // default []
})


fastifyGlobal.get('/', {
    config: {
      rateLimit: {
        max: 3,
        timeWindow: '1 minute'
      }
    }
  }, (req, reply) => {
    reply.send({ hello: 'from ... root' })
  })
```
This code is example from git, not work, request always norm.
If i use global limits WORK! But i no need global limits. 
whats a problem?

**Comment by mcollina** [comment]
Would you mind uploading a full example? Can you add a full script to reproduce?

**Comment by Svjard** [comment]
@borodadada Thanks for the bug report.

I see the issue, its with the timeWindow not being processed `ms` on a per route basis (if you set timeWindow = 60000 the above example will work fine for example) and then causing the cache to clear instantly on a 0 interval with the in-memory store. I'm sending a PR in a minute to fix the issue.

**Comment by JimKnopf182** [comment]
still not working, also with type number!?

`  fastify.post('/', {
    config: {
      rateLimit: {
        timeWindow: 60000,
        max: 6
      }
    },
    schema: { querystring: querystringSchema, params: paramsSchema },
    preHandler: [authHandler],
    handler: handler
  });`

**Comment by JimKnopf182** [comment]
ahh the problem is the authHandler, but I want to limit rate for this handler!? thanks in advance.

**Comment by mcollina** [comment]
I think that's a different bug. Specifically, we need to move the code to use a onRequest hook instead of a preHandler.

**Comment by Eomm** [comment]
This has been fixed on #97 👍
